### DISCUSSION TITLE
Query about README AluEnv code

### DISCUSSION THREAD

**Discussion by jonpovey** [pyuvm/pyuvm#33] [Q&A] [answered]
Hello, looking through the README I see this in the AluEnv code.
Is there a reason why the `getattr()` call is inside the loop, or is this just an oversight?
Can the get_method change during the run?

```python
    async def run_phase(self):
        while True:
            get_method = getattr(self.proxy, self.method_name)
            datum = await get_method()
            self.ap.write(datum)```

**Comment by raysalemi** [comment, marked as answer]
Wow.  You're right. I never noticed that.

It should definitely go into `start_of_simulation_phase()` or the like.
